libraries/ui/src/Nav.tsx

```diff
@@ -1,27 +1,142 @@
 
-type NavButtonProps = LinkOrButtonProps;
+const SearchIcon: React.FC = () => (
+  <svg xmlns="http://www.w3.org/2000/svg" width="48" height="48" viewBox="0 0 48 48" fill="none">
+    <circle cx="21.4853" cy="22.4853" r="6" transform="rotate(-45 21.4853 22.4853)" stroke="currentColor" strokeWidth="2" strokeLinecap="round" />
+    <path d="M27.1421 28.1421L30.6776 31.6776" stroke="currentColor" strokeWidth="2" strokeLinecap="round" />
+  </svg>
+);
+
+const DropdownIcon: React.FC<{ expanded: boolean }> = ({ expanded }) => (
+  <svg
+    className={clsx('size-4 transition-transform', expanded ? 'rotate-180' : '')}
+    viewBox="0 0 24 24"
+    fill="none"
+    stroke="currentColor"
+  >
+    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M19 9l-7 7-7-7" />
+  </svg>
+);
+
+const ExploreSection: React.FC<{
+  expanded: boolean;
+  innerClassName?: string;
+  className?: string;
+  courses: Array<{ title: string; href: string; isNew?: boolean }>;
+}> = ({
+  expanded, innerClassName, className, courses,
+}) => (
+  <div
+    className={clsx(
+      'nav-explore-section__content-wrapper overflow-hidden transition-[max-height,opacity] duration-300',
+      expanded ? 'max-h-96 opacity-100' : 'max-h-0 opacity-0',
+      className,
+    )}
+  >
+    <div
+      className={clsx(
+        'nav-explore-section___dropdown-content flex flex-col gap-[14px] w-fit overflow-hidden text-pretty',
+        innerClassName,
+      )}
+    >
+      <h3 className="nav-explore-section__dropdown-title font-bold">Our courses</h3>
+      {courses?.map((course) => (
+        <a key={course.href} href={course.href} className="nav-explore-section__dropdown-link">
+          {course.isNew && (
+            <span className="nav-explore-section__new-badge text-bluedot-normal font-black pr-2">
+              New!
+            </span>
+          )}
+          {course.title}
+        </a>
+      ))}
+    </div>
+  </div>
+);
+
+const NavLinks: React.FC<{
+  exploreSectionInline?: boolean;
+  children: React.ReactNode;
+  courses: Array<{ title: string; href: string; isNew?: boolean }>;
+  exploreExpanded: boolean;
+  onToggleExplore: () => void;
+  className?: string;
+}> = ({
+  exploreSectionInline, children, courses, exploreExpanded, onToggleExplore, className,
+}) => (
+  <div className={clsx('nav-links flex gap-9', className)}>
+    <div>
+      <button
+        type="button"
+        onClick={onToggleExplore}
+        className="nav-links__dropdown-button flex items-center gap-2 hover:text-bluedot-normal"
+      >
+        Explore
+        <DropdownIcon expanded={exploreExpanded} />
+      </button>
+      {exploreSectionInline && (
+        <ExploreSection
+          expanded={exploreExpanded}
+          courses={courses}
+          className="nav-links__explore-section"
+          innerClassName="pl-6 pt-6"
+        />
+      )}
+    </div>
+    {children}
+  </div>
+);
+
+const LoginButtons: React.FC<{ className?: string }> = ({ className }) => (
```

Comment:
i'd recommend a more general naming here, as currently this isn't login
```suggestion
const CTAButtons: React.FC<{ className?: string }> = ({ className }) => (
```